src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1469,6 +1479,10 @@ func (k *kataAgent) handleDeviceBlockVolume(c *Container, m Mount, device api.De
 	if len(vol.Options) == 0 {
 		vol.Options = m.Options
 	}
+	if m.FSGroup > 0 {
```

Comment:
That's a great point!
I've updated the `agent.proto` to create a `FSGroup` message

```proto
// FSGroup consists of the group id and group ownership change policy
// that a volume should have its ownership changed to.
message FSGroup {
	// GroupID is the ID that the group ownership of the
	// files in the mounted volume will need to be changed to.
	uint32 group_id = 2;
	// GroupChangePolicy specifies the policy for applying group id
	// ownership change on a mounted volume.
	types.FSGroupChangePolicy group_change_policy = 3;
}
```

and have Storage use this for the `fs_group` field. This will allow the field to be optional
and in `mount.rs`, we check if this field is set and will perform the chown on the group id.

This should cover the case of user setting fsGroup to some value and then setting it back to 0 (root) after.

Thanks for spotting this.